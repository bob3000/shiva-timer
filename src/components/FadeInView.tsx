import * as React from 'react';
import { Animated, ViewStyle } from 'react-native';

interface IFadeInViewProps {
  style?: ViewStyle;
}

interface IFadeInViewState {
  fadeAnim: Animated.Value;
}

export default class FadeInView extends React.Component<
  IFadeInViewProps,
  IFadeInViewState
> {
  constructor(props: IFadeInViewProps) {
    super(props);
    this.state = {
      fadeAnim: new Animated.Value(0),
    };
  }

  public componentDidMount() {
    Animated.timing(this.state.fadeAnim, {
      duration: 10000,
      toValue: 1,
    }).start();
  }

  public render() {
    const { fadeAnim } = this.state;

    return (
      <Animated.View
        style={{
          ...this.props.style,
          opacity: fadeAnim,
        }}
      >
        {this.props.children}
      </Animated.View>
    );
  }
}
