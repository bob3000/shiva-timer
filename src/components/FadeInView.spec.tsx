import * as React from 'react';
import * as renderer from 'react-test-renderer';
import FadeInView from './FadeInView';

jest.useFakeTimers();

describe('FadeInView', () => {
  it('snapshot test', () => {
    const tree = renderer.create(<FadeInView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
