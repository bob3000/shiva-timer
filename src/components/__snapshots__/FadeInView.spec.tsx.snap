// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeInView snapshot test 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
/>
`;
